lib/galaxy/config/__init__.py

```diff
@@ -133,37 +133,36 @@ def _set_config_directories():
                 self.config_dir = os.path.abspath(self.config_dir)
 
             self.data_dir = config_kwargs.get('data_dir')
-            # mutable_config_dir is intentionally not configurable. You can
-            # override individual mutable configs with config options, but they
-            # should be considered Galaxy-controlled data files and will by default
-            # just live in the data dir
             self.sample_config_dir = os.path.join(os.path.dirname(__file__), 'sample')
+            self.managed_config_dir = config_kwargs.get('managed_config_dir')
 
             if running_from_source:
                 if self.config_dir is None:
                     self.config_dir = os.path.join(self.root, 'config')
                 if self.data_dir is None:
                     self.data_dir = os.path.join(self.root, 'database')
-                self.mutable_config_dir = self.config_dir
+                if self.managed_config_dir is None:
+                    self.managed_config_dir = self.config_dir
             else:
                 if self.config_dir is None:
                     self.config_dir = os.getcwd()
                 if self.data_dir is None:
                     self.data_dir = os.path.join(self.config_dir, 'data')
-                self.mutable_config_dir = os.path.join(self.data_dir, 'config')
+                if self.managed_config_dir is None:
+                    self.managed_config_dir = os.path.join(self.data_dir, 'config')
 
             # TODO: do we still need to support ../shed_tools when running_from_source?
             self.shed_tools_dir = os.path.join(self.data_dir, 'shed_tools')
```

Comment:
Not sure if you like the helper methods or not, just trying to be consistent, feel free to disregard.

```suggestion
            self.shed_tools_dir = self._in_data_dir('shed_tools')
```

Reply:
Yep, consistent and better.